---
path: "/learn/userguides/data-processing-pipelines/qc-mertics"
date: "2018-05-03"
title: "Data Processing Pipelines QC Metrics"
---

## Data Processing Pipelines QC Metrics

## Overview

Here we provide the quality control metrics used to evaluate data quality for each data processing pipeline. Each metric is listed along with links to the tool used to perform that particular analysis and a description of the metric. If you want to submit a QC metric for addition to a pipeline or for feedback, please contact us at data-help@humancellatlas.org.

## Smart-seq2 Pipeline Metrics

The Smart-seq2 pipeline processes data generated from plate-based Smart-seq2 scRNA sequencing protocols (full transcript). The metrics below are generated from the quality control module of the pipeline.

| Metric                   | Program |Details | 
|-------------------------------|------------- |------------------------|
|RnaSeqMetrics |[CollectRnaSeqMetrics](https://broadinstitute.github.io/picard/command-line-overview.html#CollectRnaSeqMetrics)|[Metrics Definitions](http://broadinstitute.github.io/picard/picard-metric-definitions.html#RnaSeqMetrics) |
|DuplicationMetrics|[MarkDuplicates](https://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates)       |[Metrics Definitions](http://broadinstitute.github.io/picard/picard-metric-definitions.html#DuplicationMetrics) |
|AlignmentSummaryMetrics |[CollectMultipleMetrics](https://broadinstitute.github.io/picard/command-line-overview.html#CollectMultipleMetrics) |[Metrics Definitions](https://broadinstitute.github.io/picard/picard-metric-definitions.html#AlignmentSummaryMetrics) |
| InsertSizeMetrics |[CollectMultipleMetrics](https://broadinstitute.github.io/picard/command-line-overview.html#CollectMultipleMetrics) |[Metrics Definitions](https://broadinstitute.github.io/picard/picard-metric-definitions.html#InsertSizeMetrics)     |
|GcBiasMetrics,GcBiasDetailMetrics |[CollectMultipleMetrics](https://broadinstitute.github.io/picard/command-line-overview.html#CollectMultipleMetrics) |[Metrics Definitions](https://broadinstitute.github.io/picard/picard-metric-definitions.html#GcBiasDetailMetrics) |
| QualityYieldMetrics | [CollectMultipleMetrics](https://broadinstitute.github.io/picard/command-line-overview.html#CollectMultipleMetrics) |[Metrics Definitions](https://broadinstitute.github.io/picard/picard-metric-definitions.html#CollectQualityYieldMetrics.QualityYieldMetrics) |
|SequencingArtifactMetrics  | [CollectMultipleMetrics](https://broadinstitute.github.io/picard/command-line-overview.html#CollectMultipleMetrics) |[Metrics Definitions](https://broadinstitute.github.io/picard/picard-metric-definitions.html#ErrorSummaryMetrics) |

## Optimus Prime 10x V2

This pipeline, described [here](https://dev.data.humancellatlas.org/learn/userguides/data-processing-pipelines/optimus-prime-workflow-in-detail), processes genomic data generated from the [10x Genomics](https://www.10xgenomics.com/solutions/single-cell/) 3 prime v2 assay.  The metrics below are detected using Single Cell Tools ([sctools](https://github.com/HumanCellAtlas/sctools)).

| Cell Metrics                  | Program            |Details                 | 
|-------------------------------|--------------------|------------------------|
|n_reads |[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads associated with this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html) |
|noise_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Number of reads that are categorized by 10x Genomics Cell Ranger as "noise". Refers to long polymers, or reads with high numbers of N (ambiguous) nucleotides. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|perfect_molecule_barcodes|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads with molecule barcodes that have no errors. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_exonic|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads for this entity that are mapped to exons. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_intronic|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads for this entity that are mapped to introns. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_utr|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads for this entity that are mapped to 3' untranslated regions (UTRs). [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_uniquely|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads mapped to a single unambiguous location in the genome. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_multiple|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads mapped to multiple genomic positions with equal confidence. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|duplicate_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads that are duplicates (see README.md for definition of a duplicate). [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|spliced_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads that overlap splicing junctions. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|antisense_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads that are mapped to the antisense strand instead of the transcribed strand. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|molecule_barcode_fraction_bases_above_30_mean|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average fraction of bases in molecule barcodes that receive quality scores greater than 30 across the reads of this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|molecule_barcode_fraction_bases_above_30_variance|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The variance in the fraction of bases in molecule barcodes that receive quality scores greater than 30 across the reads of this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genomic_reads_fraction_bases_quality_above_30_mean|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average fraction of bases in the genomic read that receive quality scores greater than 30 across the reads of this entity (included for 10x Cell Ranger count comparison). [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genomic_reads_fraction_bases_quality_above_30_variance|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The variance in the fraction of bases in the genomic read that receive quality scores greater than 30 across the reads of this entity (included for 10x Cell Ranger count comparison). [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genomic_read_quality_mean|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Average quality of Illumina base calls in the genomic reads corresponding to this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genomic_read_quality_variance|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Variance in quality of Illumina base calls in the genomic reads corresponding to this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|n_molecules|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Number of molecules corresponding to this entity. See README.md for the definition of a Molecule. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|n_fragments|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Number of fragments corresponding to this entity. See README.md for the definition of a Fragment. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_per_molecule|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average number of reads associated with each molecule in this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_per_fragment|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average number of reads associated with each fragment in this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|fragments_per_molecule|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average number of fragments associated with each molecule in this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|fragments_with_single_read_evidence|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of fragments associated with this entity that are observed by only one read. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|molecules_with_single_read_evidence|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of molecules associated with this entity that are observed by only one read. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|perfect_cell_barcodes|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads whose cell barcodes contain no error. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_intergenic|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads mapped to an intergenic region for this cell. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_too_many_loci|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads that were mapped to too many loci across the genome and as a consequence, are reported unmapped by the aligner. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|cell_barcode_fraction_bases_above_30_variance|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The variance of the fraction of Illumina base calls for the cell barcode sequence that are greater than 30, across molecules. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|cell_barcode_fraction_bases_above_30_mean|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average fraction of Illumina base calls for the cell barcode sequence that are greater than 30, across molecules. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|n_genes|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of genes detected by this cell. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genes_detected_multiple_observations|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of genes that are observed by more than one read in this cell. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|


| Gene Metrics                  | Program            |Details                 | 
|-------------------------------|--------------------|------------------------|
|n_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads associated with this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|noise_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Number of reads that are categorized by 10x Genomics Cell Ranger as "noise". Refers to long polymers, or reads with high numbers of N (ambiguous) nucleotides. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|perfect_molecule_barcodes|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads with molecule barcodes that have no errors. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_exonic|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads for this entity that are mapped to exons. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_intronic|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads for this entity that are mapped to introns. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_utr|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads for this entity that are mapped to 3' untranslated regions (UTRs). [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_uniquely|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads mapped to a single unambiguous location in the genome. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_mapped_multiple|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads mapped to multiple genomic positions with equal confidence. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|duplicate_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads that are duplicates (see README.md for definition of a duplicate).  [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|spliced_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads that overlap splicing junctions. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|antisense_reads|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of reads that are mapped to the antisense strand instead of the transcribed strand. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|molecule_barcode_fraction_bases_above_30_mean|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average fraction of bases in molecule barcodes that receive quality scores greater than 30 across the reads of this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|molecule_barcode_fraction_bases_above_30_variance|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The variance in the fraction of bases in molecule barcodes that receive quality scores greater than 30 across the reads of this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genomic_reads_fraction_bases_quality_above_30_mean|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average fraction of bases in the genomic read that receive quality scores greater than 30 across the reads of this entity (included for 10x Cell Ranger count comparison). [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genomic_reads_fraction_bases_quality_above_30_variance|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The variance in the fraction of bases in the genomic read that receive quality scores greater than 30 across the reads of this entity (included for 10x Cell Ranger count comparison). [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genomic_read_quality_mean|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Average quality of Illumina base calls in the genomic reads corresponding to this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|genomic_read_quality_variance|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Variance in quality of Illumina base calls in the genomic reads corresponding to this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|n_molecules|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Number of molecules corresponding to this entity. See README.md for the definition of a Molecule. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|n_fragments|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|Number of fragments corresponding to this entity. See README.md for the definition of a Fragment. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_per_molecule|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average number of reads associated with each molecule in this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|reads_per_fragment|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average number of reads associated with each fragment in this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|fragments_per_molecule|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The average number of fragments associated with each molecule in this entity. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|fragments_with_single_read_evidence|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of fragments associated with this entity that are observed by only one read. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|molecules_with_single_read_evidence|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of molecules associated with this entity that are observed by only one read. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|number_cells_detected_multiple|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of cells which observe more than one read of this gene. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
|number_cells_expressing|[SC Tools](https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics)|The number of cells that detect this gene. [Metrics Definitions](https://sctools.readthedocs.io/en/cw-add-readthedocs-hooks/sctools.metrics.html)|
