---
path: "/contribute/overview/overview"
date: "2018-05-03"
title: "Overview"
template: "contributeOverviewTemplate"
linked:
    - ./contact-us.md
    - ./prepare-metadata.md
    - ./upload-data.md
    - ./see-and-share.md
---

# Contribute

The Human Cell Atlas is built with the help of and to the benefit of the scientific community. Our team welcomes contributions of any type of data at cellular resolution, as long as it is consented for open access to the public.

## Supported data types

1. **SmartSeq2 and 10X v2 RNA seq data**  - currently supported by our data analysis pipelines

    All submitted SmartSeq2 and 10X v2 experiments will be processed by our data processing pipelines, and the alignment, cell-gene matrix and quantification results will be made available in the portal alongside the raw data. 

1. **Single nucleus sequencing or image-based transcriptomics** - data analysis pipelines under development

    We are actively investigating standard analysis pipelines for single nucleus sequencing and several image-based transcriptomics experiments. If you are doing these types of experiments,  please reach out to us. We are interested to work together and better understand the data types and how to process them.

1. **Other data types at cellular resolution**

    If you have other data types not mentioned here, please reach out to us so we can develop a plan for incorporating these data into the HCA.

## Data contribution process

Once you are ready to contribute your data, you will be supported by the Human Cell Atlas data wranglers.  We currently use spreadsheets to collect the metadata and an upload tool to deposit the data files in our cloud infrastructure. 

<!--
insert icons here
    - ./contact-us.md
    - ./prepare-metadata.md
    - ./upload-data.md
    - ./see-and-share.md
-->

Ready to contribute? Start the conversation by emailing [data-help@humancellatlas.org](mailto:data-help@humancellatlas.org)

<!-- we need another icon here to show the analysis process. We also need to remove the links at the bottom, but introduce this text in more- actionable way - work with Dave and Fran) -->

## What to prepare

In order to be able to process your data at cellular resolution, we will need you to have this information ready:

- Unprocessed data files
- Information on samples
- Information on experiment and protocols
- Summary of your project and team information
- Confirmation that data is consented for open access

## Your data in the HCA

Once you contribute your cellular resolution data to the Human Cell Atlas, you will be able to find it in the data portal. You will also receive:

1. **Analysis output** - currently for human Smartseq2 and 10X v2 only

    For SmartSeq2 and 10X v2 single cell RNAseq experiments, the data will also be processed by our standard analysis pipelines. The analysis outputs will be available alongside the raw data and will be discoverable in the Data Browser.

    Your data will also be available for use directly in [analysis applications](https://dev.data.humancellatlas.org/analyze/methods/methods) available through the Data Portal.

1. **Repository accessions**

    To support you in publishing your data, we will submit the raw data to community standard archives and return to you the archival accessions that you need for publication.

    Currently we are able to archive any sequencing data which is submitted to us. Our default archives for submission are the BioStudies, BioSamples, and European Nucleotide Archives based at EMBL-EBI. We are working towards supporting submission to GEO if your funding mandates submission there. Please let the data wrangler know if this is the case when you start your submission process.

    At the moment we will not automatically archive imaging data. We are engaged with the community to understand what the archival best practice are for biological images, and we will work to support submission to these archives in the future.

Start the HCA submission process by emailing [data-help@humancellatlas.org](mailto:data-help@humancellatlas.org).
