---
path: "/contribute/overview/overview"
date: "2018-05-03"
title: "Overview"
noNav: true
---

# Contributing Data to the Human Cell Atlas

The Human Cell Atlas is built with the help of and for the benefit of the scientific community. Our team welcomes contributions of raw data at cellular resolution, as long as it is consented for open access to the public.

## Supported data types

1. **Single-cell Smart-seq2 and 10X v2 RNA-seq data**

    Data from Smart-seq2 and 10X v2 experiments will be analyzed by our [processing pipelines](/learn/userguides/data-processing-pipelines/overview-of-data-processing-pipelines-user-guides). Alignment and quantification results will be made available in the Data Browser alongside raw data. 

1. **Single nucleus sequencing and image-based transcriptomics data**

    We are actively investigating standard processing pipelines for single nucleus sequencing and several image-based transcriptomics experiments. If you are doing these types of experiments, please reach out to us. We are interested in working together to better understand these data types and how to process them.

1. **Other cellular resolution data**

    If you have other data types not mentioned here, please reach out to us so we can develop a plan for incorporating these data into the HCA.

## Data contribution process

Once you are ready to contribute your data, you will be supported by the Human Cell Atlas data wranglers. We currently use spreadsheets to collect the metadata and an upload tool to transfer the data files to our cloud infrastructure. 

<!--
insert icons here

    - ./contact-us.md
    - ./prepare-metadata.md
    - ./upload-data.md
    - ./see-and-share.md
-->

Ready to contribute? Start the conversation by emailing [data-help@humancellatlas.org](mailto:data-help@humancellatlas.org).

## What to prepare

In order to process you cellular resolution data, we will request the following information:

- Unprocessed data files
- Information on samples
- Information on experiment and protocols
- Summary of project and team information
- Confirmation that data is consented for open access

## Your data in the HCA

Once submitted to the Human Cell Atlas, your data will be searchable in the [Data Browser](https://dev.data.humancellatlas.org/explore/projects). You will also receive:

1. **Analysis results**

    For Smart-seq2 and 10X v2 single-cell RNA-seq experiments, data will be processed by our standard [processing pipelines](/learn/userguides/data-processing-pipelines/overview-of-data-processing-pipelines-user-guides). Alignment and quantification results will be made available in the Data Browser alongside raw data.

    Your data will also be available for use directly in [analysis applications](https://dev.data.humancellatlas.org/analyze/methods/methods/) available through the Data Portal.

1. **Repository accessions**

    To support publishing your research, we will submit raw data to community standard archives and return the archival accessions that you need for publication.

    Currently we are able to archive projects, samples, and sequencing data in BioStudies, BioSamples, and the European Nucleotide Archives (ENA), respectively. We are working towards supporting submission to the NCBI Gene Expression Omnibus (GEO) if your funding mandates submission there. Please let a data wrangler know if this is the case when you start your submission process.

    At the moment we will not automatically archive imaging data. We are engaged with the community to understand what the archival best practices are for biological images, and we will work to support submission to these archives in the future.

Start the HCA submission process by emailing [data-help@humancellatlas.org](mailto:data-help@humancellatlas.org).
